### PR TITLE
babashka: 0.0.78 -> 0.0.89

### PR DESCRIPTION
**Pull request by bennyandresen**
###### Motivation for this change
Version bump

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

@AndersonTorres

### CODE REVIEW

**Comment by bhougland18** [comment]
Thanks @bandresen! I just saw this yesterday.  This version has some big enhancements.

**Comment by FRidh** [comment]
@GrahamcOfBorg build babashka

**Comment by FRidh** [comment]
@GrahamcOfBorg build babashka

**Comment by bhougland18** [comment]
Any possibility to bump it to 0.0.8 instead?

**Comment by bandresen** [comment]
I sill don't know why @GrahamcOfBorg doesn't build this (or clj-kondo) but it compiles fine on my system.

**Comment by bandresen** [comment]
Maybe @GrahamcOfBorg doesn't try to build because graalvm8 has not been build successfully hydra in quite awhile. (Got this from a comment in #84354 by @jlesquembre

**Comment by bandresen** [comment]
for the next reviewer:

according to @hlolli in #83080 the package can't be built because it gets OOM-killed in grahamofborg's build.
I measured the peak memory resource consumption at 25GB when I built it myself.

It builds fine on my system and could be included, as the situation is the same for 0.0.78 or 0.0.71 before it.

**Comment by bandresen** [comment]
Updated to latest available babashka version at this point in time (0.0.88-2)

**Comment by bandresen** [comment]
Also pushed a commit for the utf-8 error that was found and fixed by @bfortz in #85393

**Comment by hlolli** [comment]
4wiw, I'm still working hard on getting graalvm8/11 working on v20. These statically compiled .a libraries needed to run the jvmci8 is giving me a very hard time. But I hope to have a PR open soon.

**Comment by hlolli** [comment]
Sorry not jvmci8, but substratevm expects jdk home with jvmci8 (in graal8's case) with .a libraries, and those have to come from jdk8 hotspot. With oracle using their own build farm to produce these files (ref. https://github.com/graalvm/openjdk8-jvmci-builder/issues/11 ) it makes this bit difficult path to get right.

File: pkgs/development/interpreters/clojure/babashka.nix [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
     url = "https://github.com/borkdude/${pname}/releases/download/v${version}/${pname}-${version}-standalone.jar";
-    sha256 = "01w990zk5qjrbnc846snh6na002kdyrlrfnqwg03ibx20g3mr7if";
+    sha256 = "1hv37lvpvyxq7788kd3ig90mwzr72h3za30yhk9gcwynmnga0gin";
   };
 
   dontUnpack = true;
 
-  buildInputs = [ graalvm8 ];
+  LC_ALL = "en_US.UTF-8";
+  buildInputs = [ graalvm8 glibcLocales ];
```
Comment (Mic92) [inline pkgs/development/interpreters/clojure/babashka.nix:22]: `glibcLocales` is used at build time. Hence it should be put into `nativeBuildInputs`.

Comment (Mic92) [reply pkgs/development/interpreters/clojure/babashka.nix:22]: Same applies to `graalvm8`:

```suggestion
  nativeBuildInputs = [ graalvm8 glibcLocales ];
```

Comment (bandresen) [reply pkgs/development/interpreters/clojure/babashka.nix]: Corrected, thanks!

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 84350` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- babashka
</details>

**Pull request merged**